pulser-core/pulser/register/traps.py

```diff
@@ -95,7 +95,7 @@ def get_traps_from_coordinates(self, *coordinates: ArrayLike) -> list[int]:
         """
         traps = []
         rounded_coords = np.round(
-            np.array(coordinates), decimals=COORD_PRECISION
+            np.array(coordinates).astype(float), decimals=COORD_PRECISION
```

Comment:
nit: here it would be more natural to do
```suggestion
            np.array(coordinates, dtype=float), decimals=COORD_PRECISION
```